packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -130,28 +130,77 @@ class OnsetPicker extends React.Component {
       onsetDate: this.props.onsetDate ? moment(this.props.onsetDate) : undefined,
       age: ageValues.age,
       ageUnit: ageValues.ageUnit,
+      year: undefined,
+      month: undefined,
     };
   }
 
+  componentDidMount() {
+    if (this.props.onsetDate) {
+      const firstDayOfYear = moment().year(moment(this.props.onsetDate).year()).startOf('year').format(DATE_FORMAT);
+      const firstDayOfMonth = moment().year(moment(this.props.onsetDate).year()).month(moment(this.props.onsetDate).month()).startOf('month')
```

Comment:
Can we use a variable to store `moment(this.props.onsetDate)` so that it doesn't have to create a new instance every time? And possibly use one for `moment().year(moment(this.props.onsetDate).year())` as well.

Reply:
Added Variables c4b4f2f